### PR TITLE
BRC  Mishra, Eminent One

### PR DESCRIPTION
**Pull request by Simisays**
Since I'm scripting this not at home, I haven't been able to playtest them. Though I assume they are correct

### CODE REVIEW

File: forge-gui/res/cardsfolder/upcoming/mishra_eminent_one.txt [hunk 1]
```diff
@@ -0,0 +1,10 @@
+Name:Mishra, Eminent One
+ManaCost:2 U B R
+Types:Legendary Creature Human Artificer
+PT:5/4
+T:Mode$ Phase | Phase$ BeginCombat | ValidPlayer$ You | TriggerZones$ Battlefield | Execute$ TrigCopy | TriggerDescription$ At the beginning of combat on your turn, create a token that's a copy of target noncreature artifact you control, except its name is Mishra's Warform and it's a 4/4 Construct artifact creature in addition to its other types. It gains haste until end of turn. Sacrifice it at the beginning of the next end step.
+SVar:TrigCopy:DB$ CopyPermanent | ValidTgts$ Artifact.nonCreature+YouCtrl | TgtPrompt$ Select target noncreature artifact you control. | AddTypes$ Construct | PumpKeywords$ Haste | AtEOT$ Exile | NewName$ Mishra's Warform | SetPower$ 4 | SetToughness$ 4
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/mishra_eminent_one.txt:6]: Wrong AtEOT

File: forge-gui/res/cardsfolder/upcoming/urza_chief_artificer.txt [hunk 2]
```diff
@@ -0,0 +1,12 @@
+Name:Urza, Chief Artificer
+ManaCost:3 W U B
+Types:Legendary Creature Human Artificer
+PT:4/5
+K:Affinity:Creature.Artifact+YouCtrl:artifact creatures
```
Comment (Hanmac) [inline forge-gui/res/cardsfolder/upcoming/urza_chief_artificer.txt:5]: This doesn't work yet

The Keyword needs to be updated first
https://github.com/Card-Forge/forge/blob/abf25a50f0ab8359249355ac2169ae8598c1ab93/forge-game/src/main/java/forge/game/card/CardFactoryUtil.java#L3535

File: forge-gui/res/cardsfolder/upcoming/urza_chief_artificer.txt [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+ManaCost:3 W U B
+Types:Legendary Creature Human Artificer
+PT:4/5
+K:Affinity:Creature.Artifact
+S:Mode$ Continuous | Affected$ Creature.Artifact+YouCtrl | AddKeyword$ Menace | Description$ Artifact creatures you control have menace.
+T:Mode$ Phase | Phase$ End of Turn | ValidPlayer$ You | TriggerZones$ Command | Execute$ TrigToken | TriggerDescription$ At the beginning of your end step, create a 0/0 colorless Construct artifact creature token with "This creature gets +1/+1 for each artifact you control."
+SVar:TrigToken:DB$ Token | TokenScript$ c_0_0_a_construct_total_artifacts
+SVar:PlayMain1:TRUE
+DeckHints:Type$Artifact
+DeckHas:Type$Token|Construct
```
Comment (Northmoc) [inline forge-gui/res/cardsfolder/upcoming/urza_chief_artificer.txt:11]: Ability$Token & Type$Construct

File: forge-gui/res/cardsfolder/upcoming/urza_chief_artificer.txt [hunk 4]
```diff
@@ -2,7 +2,7 @@ Name:Urza, Chief Artificer
 ManaCost:3 W U B
 Types:Legendary Creature Human Artificer
 PT:4/5
-K:Affinity:Creature.Artifact+YouCtrl:artifact creatures
+K:Affinity:Creature.Artifact
```
Comment (Northmoc) [inline forge-gui/res/cardsfolder/upcoming/urza_chief_artificer.txt:5]: should be good with `K:Affinity:Creature.Artifact:artifact creature`

YouCtrl is added automatically in CardFactoryUtil.addStaticAbility()

@Hanmac we did have to update the description stuff for Junk Winder, so I think this should be good

File: forge-gui/res/cardsfolder/upcoming/mishra_eminent_one.txt [hunk 5]
```diff
@@ -0,0 +1,10 @@
+Name:Mishra, Eminent One
+ManaCost:2 U B R
+Types:Legendary Creature Human Artificer
+PT:5/4
+T:Mode$ Phase | Phase$ BeginCombat | ValidPlayer$ You | TriggerZones$ Battlefield | Execute$ TrigCopy | TriggerDescription$ At the beginning of combat on your turn, create a token that's a copy of target noncreature artifact you control, except its name is Mishra's Warform and it's a 4/4 Construct artifact creature in addition to its other types. It gains haste until end of turn. Sacrifice it at the beginning of the next end step.
+SVar:TrigCopy:DB$ CopyPermanent | ValidTgts$ Artifact.nonCreature+YouCtrl | TgtPrompt$ Select target noncreature artifact you control. | AddTypes$ Construct | PumpKeywords$ Haste | AtEOT$ Sacrifice | NewName$ Mishra's Warform | SetPower$ 4 | SetToughness$ 4
```
Comment (Northmoc) [inline forge-gui/res/cardsfolder/upcoming/mishra_eminent_one.txt:6]: remove `.` from TgtPrompt please ;)

**Comment by Northmoc** [comment]
> Since I'm scripting this not at home, I haven't been able to playtest them. Though I assume they are correct

I'm going to mark this as WIP – I think it's important that you actually test these before merge
We will catch a lot but nothing beats actually trying them out

**Comment by Simisays** [comment]
@Northmoc Yeah tested it just now. Mishra's works fine but not Urza's affinity for artifacts

**Review by tool4ever — APPROVED** [review]

**Pull request merged**